src/common/event-utils/index.ts

```diff
@@ -1,67 +1,67 @@
+function checkModifiers(e: KeyboardEvent): boolean {
+    return e.ctrlKey || e.altKey || e.metaKey || e.shiftKey;
+}
+
 function handleKeydown(
-    keyCodes: number[],
+    keyCodes: string[],
     e: KeyboardEvent,
     callback: () => any,
 ) {
-    const keyCode = e.charCode || e.keyCode;
+    const keyCode = e.key
     if (keyCodes.indexOf(keyCode) !== -1) {
         callback();
     }
 }
 
 // inverse of found keys
 function handleNotKeydown(
-    keyCodes: number[],
+    keyCodes: string[],
     e: KeyboardEvent,
     callback: () => any,
 ) {
-    const keyCode = e.charCode || e.keyCode;
-    if (keyCodes.indexOf(keyCode) === -1) {
+    const keyCode = e.key;
+    if (keyCodes.indexOf(keyCode) === -1 && !checkModifiers(e)) {
         callback();
     }
 }
 
 // enter key
 function handleEnterKeydown(e: KeyboardEvent, callback: () => any) {
-    handleKeydown([13], e, callback);
+    handleKeydown(["Enter"], e, callback);
 }
 
 // space and enter keys
 function handleActionKeydown(e: KeyboardEvent, callback: () => any) {
-    handleKeydown([32, 13], e, callback);
+    handleKeydown(["Enter", " "], e, callback);
 }
 
 function handleEscapeKeydown(e: KeyboardEvent, callback: () => any) {
-    handleKeydown([27], e, callback);
+    handleKeydown(["Escape"], e, callback);
 }
 
 function handleUpDownArrowsKeydown(e: KeyboardEvent, callback: () => any) {
-    handleKeydown([38, 40], e, callback);
+    handleKeydown(["ArrowUp", "ArrowDown"], e, callback);
 }
 
 function handleLeftRightArrowsKeydown(e: KeyboardEvent, callback: () => any) {
-    handleKeydown([37, 39], e, callback);
+    handleKeydown(["ArrowLeft", "ArrowRight"], e, callback);
 }
 
 function handleArrowsKeydown(e: KeyboardEvent, callback: () => any) {
-    handleKeydown([37, 38, 39, 40], e, callback);
+    handleKeydown(["ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight"], e, callback);
 }
 
 // only fire for character input, not modifier/meta keys (enter, escape, backspace, tab, etc.)
 function handleTextInput(e: KeyboardEvent, callback: () => any) {
     const keys = [
-        9, // tab
-        13, // enter key
-        16, // shift
-        17, // control
-        18, // alt
-        20, // caps lock
-        27, // escape
-        37, // left arrow
-        38, // up arrow
-        39, // right arrow
-        40, // down arrow
-        91, // "meta" key (Mac "command" key)
+        "Tab",
+        "Enter",
+        "Shift",
+        "Escape",
+        "ArrowUp",
+        "ArrowDown",
+        "ArrowLeft",
+        "ArrowRight"
```

Comment:
Was it intentional to remove modifier keys here (ctrl, alt, caps lock, cmd)?

Reply:
I moved them into the function itself since the recommended way to check for modifiers is using e.shiftKey
See check modifiers
To add this function is actually not being used anywhere except here. I was considering to remove it and put the check directly inline

Reply:
This function is used in `ebay-combobox`, where `checkModifiers` is _not_ called. After your changes, `on-input-change` is emitted every time the user presses `ctrl`, `option`, `cmd` (previously that was not the case)

Reply:
So as i mentioned i added a check in handleNotKeydown where this list is being passed. I added a check to see no modiferiers is pressed.

Reply:
Sure, but I still see events emitted where they were not before in `ebay-combobox` storybook